src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQQueueClient.cs

```diff
@@ -144,53 +141,20 @@ public RabbitMQQueueClient(ConnectionFactory connectionFactory, string queueName
             Func<IQueueMessage, CancellationToken, Task> messageHandler,
             MessageHandlerOptions messageHandlerOptions)
         {
+            // Reconfigure QoS in order to support specified level of concurrency.
+            ConfigureBasicQos(prefetchCount: messageHandlerOptions.MaxConcurrentCalls);
+
             var consumer = new EventingBasicConsumer(_channel);
 
-            consumer.Received += (channel, eventArgs) =>
+            // The use of an `async void` event handler allows the calling consumer
+            // instance to continue and return before the message handler completes,
+            // which is essential for handling multiple messages concurrently with a
+            // single consumer, albeit at the expense of message processing order.
+            // Any exceptions propagating from the message handler will effectively
+            // be ignored.
+            consumer.Received += async (channel, eventArgs) =>
```

Comment:
The approach here was the simplest change to unblock the RabbitMQ consumer / channel thread, so that we can support handling multiple messages concurrently.